libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -258,6 +284,32 @@ void AP_UAVCAN::loop(void)
             continue;
         }
 
+        for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
```

Comment:
I'd like to see this broken out into a separate function, just so that loop() doesn't grow too unwieldy

Reply:
will do